### ISSUE TITLE
Prior reservation, args.prior_loss_weight doesn't exist

### ISSUE THREAD

**Issue by MV799** [d8ahazard/sd_dreambooth_extension#5]
When I use prior reservation, it tells me 
args.prior_loss_weight doesn't exist in the train_dreambooth.py
line 796
loss = loss + args.prior_loss_weight * prior_loss

I quickly turned it into loss = loss + prior_loss just to test it and it seems to be churning well. I'm not sure where args.prior_loss_weight should be defined. 

also it would be good to specify the labels correctly. 
db_train_batch_size = gr.Number(label="Batch Size", precision=1, value=1)
db_sample_batch_size = gr.Number(label="Class Batch Size", precision=1, value=1)
it's confusing, I needed to look into the code to see what is what. so Batch size should be labeled Train Batch Size etc...

**Comment by d8ahazard** [comment]
> When I use prior reservation, it tells me args.prior_loss_weight doesn't exist in the train_dreambooth.py line 796 loss = loss + args.prior_loss_weight * prior_loss
> 
> I quickly turned it into loss = loss + prior_loss just to test it and it seems to be churning well. I'm not sure where args.prior_loss_weight should be defined.
> 
> also it would be good to specify the labels correctly. db_train_batch_size = gr.Number(label="Batch Size", precision=1, value=1) db_sample_batch_size = gr.Number(label="Class Batch Size", precision=1, value=1) it's confusing, I needed to look into the code to see what is what. so Batch size should be labeled Train Batch Size etc...

This should have been fixed last night, at least the missing prior loss weight value.

As far as "confusing labels" go, I'll be writing up a readme for this today so it's not so esoteric. The problem with coming up with labels is that even the ones within the project don't necessarily line up with various terminology. For example, in one spot, the images used with Prior Preservation are referred as classifier images, but in other parts of the code, they're called "sample" images.

**Comment by MV799** [comment]
Thanks! Do you think you can maybe add the formula (that is somehow tested to work well) or maybe just put it in the labels
    Number of subject images (instance) = N
    Number of class images (regularization) = N x 12
    Maximum number of Steps = N x 80 (this is what I'm tweaking right now but between 80 and 100 should be enough)
    Learning rate = 1e-6
    Learning rate schedule = polynomial
    Learning rate warmup steps = Steps / 10

Also the default for sample_batch_size=4 in the repo (you have it as 1)

It successfully finished last night, but with the default (and about ) 100x images, it brutally overtrained. I'm not sure where was the issue, I'll have to check.

**Comment by d8ahazard** [comment]
> Thanks! It successfully finished last night, but with the default and about steps 100x images, it brutally overtrained. I'm not sure where was the issue, I'll have to check. I'll try to do another batch today with the new pull. Thanks, for making this - in general, it seems to be working well.

My pleasure.

Try ensuring you have a class prompt and around 100 for the class images as well, see if that helps with the overtraining.

Closing.
